### ISSUE TITLE
Infinite loop in web-server.js

### ISSUE THREAD

**Issue by fourpastmidnight** [joeeames/AngularFundamentalsFiles#10]
While going through this course on PluralSight, I've triple-checked my code with what was presented in the Routing Module when enabling HTML5 routing (`$locationProvider.html5Mode(true);`).

Navigating to the New Event page works. Listing all the events works. I even successfully completed one of the "extra exercises" at the end of the module&mdash;adding a route for the Edit Profile page. However, no matter what I do, trying to view a single event after enabling HTML5 routing (e.g. navigate to `event/1` or even `/event/1`, for example) ends up putting the Node.js webserver into an infinite loop, eventually consuming all memory available to the Chrome (v39.x) process and crashing chrome.

I confirmed that this is the case by turning off HTML5 routing and replacing the `#` in the `href`s and everything works again.

**Comment by justinrouleau** [comment]
Probably a little late, but hopefully this helps someone else.

I was getting the same infinite loop until I changed the templateUrl for each route to include the forward slash:

    $routeProvider.when('/newEvent',
        {
            templateUrl: '/templates/NewEvent.html',
            controller: 'EditEventController'
        });

Turns out, you don't need it for all of the routes... If you look close, you'll notice that he has added a forward slash to the templateUrl for the /event/:eventId route when the Enabling HTML5 Routing clip begins.

Without the forward slash, angular is looking for the EventDetails.html template under localhost/event/templates rather than localhost/templates.
